### PR TITLE
Resolve non-unique name issue

### PR DESCRIPTION
**Pull request by mathidachuk**
A solution came to me during a game of LoL. Granted it's not the most elegant but it resolves the awful issue with duplicated names :)

### CODE REVIEW

File: R/hp_datatree2.R [hunk 1]
```diff
@@ -2,37 +2,9 @@
...
-# acme$Research$`New Product Line`$p <- 0.25
-# acme$Research$`New Labs`$p <- 0.9
-# acme$IT$Outsource$p <- 0.2
-# acme$IT$`Go agile`$p <- 0.05
-# acme$IT$`Switch to R`$p <- 1
-#
-# acme$IT$Outsource$AddChild("India")
-# acme$IT$Outsource$AddChild("Poland")
-#
+# data(acme)
```
Comment (mathidachuk) [inline R/hp_datatree2.R:5]: can just use acme dataset instead of the wall of text :D

File: R/hp_datatree2.R [hunk 2]
```diff
@@ -60,25 +32,32 @@ dt_collect <- function(x) {
...
+
+
   root <- dt_root_df(.data)
   df <- vctrs::vec_rbind(dt_collect(.data), root)
 
+
   # should only expect a single NA in the link column, this is NA comes
   # from binding the root dataframe to the tree. If there are multiple
   # NAs, then the there are missing links
-  missing_links <- "TRUE" %in% names(table(is.na(df[[link]])))
```
Comment (mathidachuk) [inline R/hp_datatree2.R:70]: do you want to talk about this lol

File: R/hp_datatree2.R [hunk 3]
```diff
@@ -91,7 +70,13 @@ hp_datatree <- function(.data, title = "Hierplane", attributes = NULL, link = "t
   }
 
   df[[link]] <- ifelse(is.na(df[[link]]), root$from, df[[link]])
-  print(df)
+
+  # Deal with duplicated child_id
+  df$child <- gsub(".*[/]", "", df[["to"]])
+  # change all the `----` back to `/`
+  df$from <- gsub("~~~placeholder~~~", "/", df$from)
```
Comment (mathidachuk) [inline R/hp_datatree2.R:77]: dont hate the placeholder

File: R/hp_datatree2.R (hunk 2 above)
Comment (tyluRp) [reply R/hp_datatree2.R:70]: Haha, yes... I forget about summing logicals, it still feels unnatural to me tho.

**Pull request merged**